apiserver/facades/agent/deployer/deployer.go

```diff
@@ -156,3 +168,48 @@ func getAllUnits(st *state.State, tag names.Tag) ([]string, error) {
 	}
 	return nil, errors.Errorf("cannot obtain units of machine %q: %v", tag, watch.Err())
 }
+
+// Remove removes every given unit from state, calling EnsureDead
+// first, then Remove. It will fail if the unit is not present.
```

Comment:
Mention the possible error types to return and have tests covered.

Reply:
The errors are just added to the ErrorResults struct. As with other facade calls, any error return is due to something unexpected. Domain logic errors due to expectation violations etc are added to the result struct abd not returned directly from the method.

Reply:
The code and method comment were copied unchanged from the common remove method. This was available on the deployer api struct directly via embedding the common remove struct, and there's an existing test for it already in the deployer test suite.

Reply:
I think the "It will fail..." comment that was copied is perhaps misleading. Any errors in removing are added to the error result struct. The comment itself describes the remove behaviour and really is relevant to the service method which is already commented. I'll remove the comment from here and the other facsdes.